File: ProgrammingBasicsJava/Jan2020/NestedLoopsLab/src/_04SumOfTwoNumbers.java
import java.util.Scanner;

public class _04SumOfTwoNumbers {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int startNum = Integer.parseInt(scanner.nextLine());
        int endNum = Integer.parseInt(scanner.nextLine());
        int magicNum = Integer.parseInt(scanner.nextLine());
        int combinationCount = 0;
        boolean isCombinationFound = false;

        for (int x = startNum; x <= endNum; x++) {
            for (int y = startNum; y <= endNum; y++) {
                combinationCount++;

                if (x + y == magicNum) {
                    System.out.printf("Combination N:%d (%d + %d = %d)",
                            combinationCount, x, y, magicNum);
                    isCombinationFound = true;
                    break;
                }
            }
            if (isCombinationFound) {
                break;
            }
        }

        if (!isCombinationFound) {
            System.out.printf("%d combinations - neither equals %d", combinationCount, magicNum);
        }
    }
}

File: JavaScriptBasics/06_1_NestedLoopsLab/04_SumOfTwoNumbers.js
function printFirstCombinationOfNumbers(input) {
    let start = Number(input[0]);
    let end = Number(input[1]);
    let magicNum = Number(input[2]);
    let countCombinations = 0;
    let isCombinationEqualtoMagicNum = false;

    for (let i = start; i <= end; i++) {
        for (let j = start; j <= end; j++) {
            countCombinations++;
            let result = i + j;

            if (result === magicNum) {
                console.log(`Combination N:${countCombinations} (${i} + ${j} = ${result})`);
                isCombinationEqualtoMagicNum = true;
                break;
            }
        }

        if (isCombinationEqualtoMagicNum === true) {
            break;
        }
    }

    if (isCombinationEqualtoMagicNum === false) {
        console.log(`${countCombinations} combinations - neither equals ${magicNum}`);
    }
}

printFirstCombinationOfNumbers(["1",
"10",
"5"]);